client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => (boxCounts[line.box]++ % BOX_PAGE_SIZE) < BOX_PAGE_SIZE)
```

Comment:
Ah yes, I guess that would work, because it's adding until it gets to a multiple of hte page size and then stopping. Happy with that. I just couldn't think of a way around it, I think I had been thinking about it too much :-P